packages/client/src/objectSet/createObjectSet.ts

```diff
@@ -143,3 +157,48 @@ export function createObjectSet<Q extends ObjectOrInterfaceDefinition>(
     },
   });
 }
+
+export function createBaseObjectSet<
+  Q extends ObjectOrInterfaceDefinition,
+>(
+  objectType: Q,
+  clientCtx: ClientContext<any>,
+  objectSet: WireObjectSet = {
+    type: "base",
+    objectType: objectType["apiName"] as string,
+  },
+): BaseObjectSet<Q> {
+  return {
+    ...createObjectSet(objectType, clientCtx, objectSet),
+
+    get: (isObjectTypeDefinition(objectType)
+      ? async <A extends SelectArg<any>>(
```

Comment:
Why can't you give` SelectArg<Q>`?

Reply:
Or maybe why don't you need to give it Q

Reply:
Yeah `Q` is probably more correct. It gets shadowed by the `BaseObjectSet` definition.